Code/Legacy/CryCommon/IAudioSystem.h

```diff
@@ -578,43 +578,32 @@ namespace Audio
     namespace Gem
     {
         ///////////////////////////////////////////////////////////////////////////////////////////////
-        // Used for initializing and releasing the audio system (AudioSystem/ATL) code.
-        class AudioSystemGemRequests
+        // Audio initialization loads resources that may not be ready until after
+        // component application (and AP) is fully up, so it is delayed until CSystem::Init.
+        // Similarly, release is called in CSystem::ShutDown.
+        class SystemBusInterface
```

Comment:
Well the ComponentLifecycle "CriticalAssetsComponent" event can be used to make sure the AP has processed assets.

There would be no dependency on CSystem::Init.

[Here](https://github.com/o3de/o3de/blob/development/Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp#L150-L158) is how the Atom Bootstrap gem module uses that event to Initialize a window

Reply:
What is considered to be a "critical asset"?  Does it still depend on the `JobDescriptor::m_critical` flag in the builder?  In terms of startup, when the application waits on `CompileCriticalAssets`, is it going to return after all `m_critical` jobs have run, or when all jobs are done?

This is the direction I want to go eventually.  Can even go further and do most of the audio initialization early, but leave the loading of assets until after we get a signal from AP that things are ready.